src/app.js

```diff
@@ -404,6 +412,22 @@ app.useAsync(async (err, req, res, next) => {
     return next(err);
   }
 
+  /* Read the entire request body (discarding it) if the request might have a
+   * body and wasn't made with Expect: 100-continue, or if it was and we wrote
+   * a 100 Continue response but then ended up here.  This ensures that the
+   * request is finished being sent before we return a (error) response, which
+   * some clients require (such as Heroku's routing/proxy layer and the Python
+   * "requests" package, but notably not curl).
+   */
+  const mayHaveBody = !["GET", "HEAD", "DELETE", "OPTIONS"].includes(req.method);
+
+  if (mayHaveBody && (!req.expectsContinue || res.wroteContinue)) {
+    const reqFinished = streamFinished(req);
+    req.unpipe();
+    req.resume();
```

Comment:
I am looking through [Stream docs](https://nodejs.org/api/stream.html) since this is new to me. What's the purpose of these two lines?

#### `req.unpipe()`

[doc](https://nodejs.org/api/stream.html#readableunpipedestination):

> If the destination is not specified, then all pipes are detached

Where were any attached?

#### `req.resume()`

[doc](https://nodejs.org/api/stream.html#readableresume):

> can be used to fully consume the data from a stream without actually processing any of that data

This one makes sense to me and is described by the block comment.

Reply:
Ah, since this is the app's global error handler, we don't really "know" what may or may not have previously called `req.pipe(…)`. It seemed safest to `.unpipe()` before resuming the stream so that we didn't resume to previous destinations.